src/test/movie.test.js

```diff
@@ -0,0 +1,35 @@
+import { screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { renderWithProviders } from './utils'
+import App from '../App'
+
+it('movies starred and saved to watch later', async () => {
+    renderWithProviders(<App />)
+
+    await userEvent.type(screen.getByTestId('search-movies'), 'forrest gump')
```

Comment:
Optimizing integration tests for speed and reliability is crucial. Using mocks for API calls can greatly improve the test performance and reduce dependencies on external services.

Consider implementing a **mocking library** like msw (Mock Service Worker) or jest.mock for mocking API calls. By simulating responses from the server without actual network requests, you'll isolate the component being tested and make the test suite more efficient and stable. This also allows you to control specific scenarios and responses, ensuring thorough testing of various conditions.